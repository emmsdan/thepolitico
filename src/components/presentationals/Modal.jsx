import React, { useState } from 'react';
import PropTypes from 'prop-types';

const Modal = props => {
  const [isModalDisplay, setIsModalDisplay] = useState('flex');
  let stylux = {
    display: isModalDisplay,
  };

  return (
    <div id={props.modal.id} className="modal" style={stylux}>
      <div className={`modal-content ${props.modal.class}`}>
        <h3>{props.modal.header}</h3>
        {props.modal.content}
        <br />
        <br />
        <button
          className="blue-gray button-bigger close"
          onClick={() => setIsModalDisplay('none')}
        >
          close
        </button>
      </div>
    </div>
  );
};

Modal.propTypes = {
  modal: PropTypes.shape({
    id: PropTypes.string,
    class: PropTypes.string,
    header: PropTypes.any,
    content: PropTypes.any,
    state: PropTypes.string,
  }),
};

export default Modal;
