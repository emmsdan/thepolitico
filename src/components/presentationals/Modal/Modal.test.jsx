import React from 'react';
import { mount } from 'enzyme';

import Modal from './Modal';

describe('<Modal />', () => {
  const modalContent = {
    id: 'AwesomeModal',
    header: 'Sample Modal header',
    content: 'Hi, I am a Modal',
    className: 'deleteModal',
    action: {
      button: {
        text: 'delete',
        className: 'red',
      },
      onClick: jest.fn(),
    },
  };

  it('should render correctly', () => {
    const component = mount(<Modal {...modalContent} />);
    expect(component).toMatchSnapshot();
  });

  it('should simulate close Modal correctly', () => {
    const component = mount(<Modal {...modalContent} />);
    const onclick = component.find('.deleteModal');
    console.log(onclick.simulate('click'))
    expect(modalContent.action.onClick).toHaveBeenCalledTimes(1);
  });
});
