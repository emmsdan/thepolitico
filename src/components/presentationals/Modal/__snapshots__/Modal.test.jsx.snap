// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> should render correctly 1`] = `
<Modal
  action={
    Object {
      "button": Object {
        "className": "red",
        "text": "delete",
      },
      "onClick": [MockFunction],
    }
  }
  className="deleteModal"
  content="Hi, I am a Modal"
  header="Sample Modal header"
  id="AwesomeModal"
>
  <div
    className="modal"
    id="AwesomeModal"
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <div
      className="modal-content deleteModal"
    >
      <h3>
        Sample Modal header
      </h3>
      <div>
         
        Hi, I am a Modal
         
      </div>
       
      <br />
      <button
        className="blue-gray button-bigger red close"
        onClick={[Function]}
      >
        delete
      </button>
      <button
        className="blue-gray button-bigger close right"
        onClick={[Function]}
      >
        close
      </button>
    </div>
  </div>
</Modal>
`;
